spec/daru/io/importers/csv_spec.rb

```diff
@@ -1,7 +1,75 @@
-require "spec_helper"
+require 'spec_helper'
+require 'webmock/rspec'
+require 'open-uri'
 
 RSpec.describe Daru::IO::Importers::CSV do
-  it "parses CSV files with format-1" do
-    expect(Daru::IO::Importers::CSV.load).to eq("CSVHelper#manipulate called by CSV#load")
+  context "loads from a CSV file" do
+    let(:path) { 'spec/fixtures/csv/matrix_test.csv' }
+    let(:col_sep) { ' ' }
+    let(:headers) { true }
+    let(:df) { Daru::IO::Importers::CSV.load(path, col_sep: col_sep, headers: headers) }
+    let('df.vectors') { [:image_resolution, :mls, :true_transform].to_index }
+    subject { df }
+
+		it { is_expected.to be_an(Daru::DataFrame) }
+    its('image_resolution.first') { is_expected.to eq(6.55779)}
+    its('true_transform.first') { is_expected.to eq("-0.2362347,0.6308649,0.7390552,0,0.6523478,-0.4607318,0.6018043,0,0.7201635,0.6242881,-0.3027024,4262.65,0,0,0,1")}
+  end
+
+  context "works properly for repeated headers" do
+    let(:path) { 'spec/fixtures/csv/repeated_fields.csv' }
+    let(:header_converters) { :symbol }
+    subject { Daru::IO::Importers::CSV.load(path, header_converters: header_converters) }
+
+		it { is_expected.to be_an(Daru::DataFrame) }
+    its('vectors.to_a') { is_expected.to eq(["id", "name_1", "age_1", "city", "a1", "name_2", "age_2"])}
+    its('age_2') { is_expected.to eq(Daru::Vector.new([3, 4, 5, 6, nil, 8]))}
+  end
+
+  context "accepts scientific notation as float" do
+    let(:path) { 'spec/fixtures/csv/scientific_notation.csv' }
+    let(:order) { ['x', 'y'] }
+    subject(:ds) { Daru::IO::Importers::CSV.load(path, order: order) }
+
+		it { is_expected.to be_an(Daru::DataFrame) }
+    its('vectors.to_a') { is_expected.to eq(['x', 'y'])}
+
+    # SPOILER ALERT : If a better syntax is possible without naming the subject,
+    # feel free to suggest / adopt it.
+    #
+    # Signed off by @athityakumar on 31/05/2017 at 10:25PM
```

Comment:
Something like (check for actual matchers, I am showing the idea):
```ruby
subject { df['y'] }
it { is_expected.to match_array(be_withing(0.001).of(9.629587310436753e+127), be_within(0.001).of(1.9341543147883677e+129), ....) }
```

Another idea is to define your own matcher named like `match_float_array` and use it here.